packages/data-context/src/sources/migration/utils.ts

```diff
@@ -289,6 +289,25 @@ export function moveSpecFiles (projectRoot: string, specs: SpecToMove[]) {
   })
 }
 
+export async function cleanupIntegrationFolder (projectRoot: string) {
+  const integrationPath = path.join(projectRoot, 'cypress', 'integration')
+  let dirContents: string[] = []
+
+  try {
+    dirContents = await fs.readdir(integrationPath)
```

Comment:
Should not need `try/catch`:

```ts

if (!fs.existsSync(dir)) {
  return
}

const dirContents = await fs.readdir(integrationPath)
// ... the rest ...

Reply:
@lachlan, don't do anything synchronous when dealing with files or you'll end up with race conditions and interlocks...

try catch is the recommended solution by nodejs

Reply:
https://nodejs.org/api/fs.html#fsexistspath-callback

Reply:
But you would probably want to use the version of fs that is in the ctx for consistency.

Passing it as a param to the function would do the trick.

Reply:
Ok then, how about `if (! (await fs.exists(dir))`? Same thing.

I don't see any obvious risk of a race condition here, though, even w/ `existsSync`.

Reply:
actually according to the docs, this is the way to deal with it:

```js
  try {
    await fs.rmdir(integrationPath)
  } catch (e: any) {
    // only throw if the folder exists
    if (e.code !== 'ENOENT') {
      throw Error(`Failed to remove ${integrationPath}`)
    }
  }
```

Reply:
oh awesome, this is much preferred. It also drives me nuts how much throws with fs and how it isn't really documented